CommandPattern.java

```diff
@@ -0,0 +1,132 @@
+package Ex4;
+import java.util.Scanner;
+public class CommandPattern {
+	public static void main(String[] args) {
+		UniversalRemote universalRemote=new UniversalRemote();
+		
+		Tv tv=new Tv();
+		SetTopBox sbox=new SetTopBox();
+		SoundSystem ss=new SoundSystem();
+		VGame vgame=new VGame();
+		
+		Command newscommand=new NewsChannelCommand(tv, sbox, ss, vgame);
+		Command serialcommand=new SerialChannelCommand(tv, sbox, ss, vgame);
+		Command ttgamecommand=new TTGameCommand(tv, sbox, ss, vgame);
+		
+		universalRemote.assignCommand(1, newscommand);
+		universalRemote.assignCommand(2, serialcommand);
+        universalRemote.assignCommand(3, ttgamecommand);
+        
+        Scanner scan=new Scanner(System.in);
+        System.out.println("Enter the Command that need to be executed..");
+        int input=scan.nextInt();
+		universalRemote.executeCommand(input);
+	}
+}
+class UniversalRemote{
+	Command comm[]=new Command[5];
+	public UniversalRemote() {
+		for(int i=0;i<5;i++) {
+			comm[i]=new DummyCommand();
+		}
+	}
+	
+	public void executeCommand(int slot) {
+		comm[slot].execute();
+	}
+	
+	public void assignCommand(int slot,Command command) {
+		comm[slot]=command;
+	}
+}
+abstract class Command{
+	Tv tv;SetTopBox sbox;SoundSystem ss;VGame vgame;
+	public Command() {
+		
+	}
+	
+	public Command(Tv tv, SetTopBox sbox, SoundSystem ss, VGame vgame) {
+		this.tv = tv;
+		this.sbox = sbox;
+		this.ss = ss;
+		this.vgame = vgame;
+	}
+	public abstract void execute();
+}
+class DummyCommand extends Command{
```

Comment:
Try to have space between class name and braces

Reply:
OK bro.